### PR TITLE
Patches picking up deployed body bags (+ resets offsets on deploy)

### PR DESCRIPTION
**Pull request by BatElite**
[major][minor]
## About the PR 
This PR fixes an oversight where dragging a deployed body bag onto your HUD lets you pick one up in its deployed state. 
There's quite a added checks: The parent call includes a QOL tile-to-tile drag I wanted to keep, and I figured dragging to HUD should try to pick up the bag instead of doing nothing (It will just open a closed bag without picking up). The others are more or less sanity checks.

I also made it so the sprite offsets get reset if you deploy a body bag, so the visual effect with the layers doesn't break if you threw it around before deploying

## Why's this needed? 
Some nerds figured out you could pick up shit including *other players* using body bags aaaaaa

### CODE REVIEW

File: code/modules/medical/surgery_tools.dm [hunk 1]
```diff
@@ -1148,8 +1150,18 @@ CONTAINS:
...
+		if(isturf(over_object))
+			..() //Lets it do the turf-to-turf slide
+			return
+		else if (istype(over_object, /obj/screen/hud))
+			MouseDrop(usr) //Try to fold & pick up the bag instead
+			return
+		else if (!ismob(over_object))
+			return
 		..()
+		var/mob/user = over_object
```
Comment (Sovexe) [inline code/modules/medical/surgery_tools.dm:1164]:
```suggestion
		else if (istype(over_object, /obj/screen/hud))
			..(usr)
		else if (over_object == usr)
			..()
		else
			return

```
Feel like emulating a new mousedrop is kind of weird, we can just make use of this one still

File: code/modules/medical/surgery_tools.dm [hunk 2]
```diff
@@ -1148,8 +1150,18 @@ CONTAINS:
...
 		if (!(src.contents && src.contents.len) && (usr == user && !usr.restrained() && !usr.stat && in_range(src, usr) && !issilicon(usr)))
```
Comment (Sovexe) [inline code/modules/medical/surgery_tools.dm:1165]:
```suggestion
		if (!length(src.contents) && usr.can_use_hands() && isalive(usr) && in_range(src, usr) && !issilicon(usr))
```
you didn't touch this line but holy shit this is gross. At least this is a little better, also removed some pointless checks

**Review by Sovexe — CHANGES_REQUESTED** [review]

File: code/modules/medical/surgery_tools.dm (hunk 2 above)
Comment (Sovexe) [inline code/modules/medical/surgery_tools.dm:1165]:
```suggestion
		if (!length(src.contents) && usr.can_use_hands() && isalive(usr) && in_range(src, usr) && !issilicon(usr))
```
you didn't touch this line but holy shit this is gross. At least this is a little better, also removed some pointless checks and we don't need that var anymore 
```

File: code/modules/medical/surgery_tools.dm [hunk 3]
```diff
@@ -1148,8 +1150,18 @@ CONTAINS:
...
+	MouseDrop(atom/over_object)
+		if (!over_object) return
+		if(isturf(over_object))
+			..() //Lets it do the turf-to-turf slide
+			return
+		else if (istype(over_object, /obj/screen/hud))
+			MouseDrop(usr) //Try to fold & pick up the bag instead
+			return
+		else if (!ismob(over_object))
+			return
```
Comment (Sovexe) [inline code/modules/medical/surgery_tools.dm:1162]:
```suggestion
		else if (istype(over_object, /obj/screen/hud))
			..(usr)
		else if (over_object == usr)
			..()
		else
			return

```
Feel like emulating a new mousedrop is kind of weird, we can just make use of this one still

**Review by Sovexe — CHANGES_REQUESTED** [review]
reposting this review as I missed up the line references

File: code/modules/medical/surgery_tools.dm (hunk 3 above)
Comment (Sovexe) [inline code/modules/medical/surgery_tools.dm:1162]:
```suggestion
		else if (istype(over_object, /obj/screen/hud))
			..(usr)
		else if (over_object == usr)
			..()
		else
			return

```
Feel like emulating a new mousedrop is kind of weird, we can just make use of this one still

Comment (BatElite) [reply code/modules/medical/surgery_tools.dm]: I did it a little differently as ```..(usr)``` didn't solve the bug (It's making the action bar on line 655 of the parent proc causing the pickup bug IIRC), but I did get rid of the mousedrop.

Comment (Sovexe) [reply code/modules/medical/surgery_tools.dm]: Curious how you got it to bug with my changes considering you're doing the exact same thing by setting `over_object` to `usr` and passing it down to the parent. I am likewise unable to reproduce the issue with my suggested code or your current code they both behave identically in my testing

**Review by Sovexe — APPROVED** [review]

**Pull request merged**